LanguageFeatures/Patterns/Exhaustiveness/guard_A01_t02.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion To tell if the set of cases in a switch statement or expression
+/// are exhaustive over the matched value type:
+/// 1. Lift the matched value type to a space union value.
+/// 2. Discard any cases that have guards. Since static analysis can't tell when
+///   a guard might evaluate to false, any case with a guard doesn't reliably
+///   match values and so can't help prove exhaustiveness.
+/// 3. Lift the remaining case patterns to a set of space unions cases.
+/// 4. The switch is exhaustive if is-exhaustive with value and cases is true
+///   and false otherwise.
+///
+/// @description Check that any cases with guards are ignored when calculating
+/// an exhaustiveness
```

Comment:
Perhaps say that exhaustiveness can be recognized based on the cases without guards, even in the case where cases with guards exist as well.